src/oop/special_class_method/medium/q10/MainTest.java

```diff
@@ -0,0 +1,275 @@
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q10." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided %s attribute. Please revert it to the original state.";
+        assertTrue(testClass.hasField(attributeName1, int.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName1));
+        assertTrue(testClass.hasField(attributeName2, String.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName2));
+        Object[][] tests = new Object[][]{
+                {3, "Red"},
+                {10, "Black"},
+                {12, "Blue"},
+                {43, "White"},
+                {7, "Yellow"}
+        };
+        modifiedClassMessage =
+                "You have modified the provided %s constructor. Please revert it to the original state.";
+        for (int i = 0; i < tests.length; i++) {
+            int size = (int) tests[i][0];
+            String colour = (String) tests[i][1];
+            Object[][] arguments = {
+                    {size, int.class},
+                    {colour, String.class}
+            };
+            Class<?>[] classes = {int.class, String.class};
+            Object testInstance = testClass.createInstance(arguments);
+            _assertEquals(size, testClass.getFieldValue(testInstance, attributeName1),
+                    String.format(modifiedClassMessage, className));
+            _assertEquals(colour, testClass.getFieldValue(testInstance, attributeName2),
+                    String.format(modifiedClassMessage, className));
+            assertTrue(testClass.hasModifier(classes, "public"),
+                    String.format(modifiedClassMessage, className));
+        }
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red"),
+                Arguments.of(10, "Black"),
+                Arguments.of(12, "Blue"),
+                Arguments.of(43, "White"),
+                Arguments.of(7, "Yellow")
+        );
+    }
+
+    private static Stream<Arguments> updateIntegerInputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red", 5),
+                Arguments.of(10, "Black", 12),
+                Arguments.of(12, "Blue", 10),
+                Arguments.of(43, "White", 37),
+                Arguments.of(7, "Yellow", 4)
+
+        );
+    }
+
+    private static Stream<Arguments> updateStringInputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red", "White"),
+                Arguments.of(10, "Black", "Green"),
+                Arguments.of(12, "Blue", "Red"),
+                Arguments.of(43, "White", "Gray"),
+                Arguments.of(7, "Yellow", "Orange")
+
+        );
+    }
+
+    @Test
+    public void getSizeIsDefinedCorrectly() {
```

Comment:
do we need these "isDefinedCorrectly" tests if we test the same things in the "correctMethod" tests?

Reply:
You can feel free to remove these going forward, as long as you test it in the "correctMethod" test.